### ISSUE TITLE
QR Code generation results in a faulted qr code

### ISSUE THREAD

**Issue by razfazz** [alexzhirkevich/custom-qr-generator-flutter#1]
the following code 

```
CustomPaint(

                    size: const Size(300, 300),
                    painter: QrPainter(
                        data:  'https://youtube.com',
                        options: const QrOptions(
                            /*shapes: QrShapes(
                                darkPixel:
                                QrPixelShapeRoundCorners(cornerFraction: .5),
                                frame:
                                QrFrameShapeRoundCorners(cornerFraction: .25),
                                ball:
                                QrBallShapeDefault()),
                            colors: QrColors(
                                dark: QrColorLinearGradient(colors: [
                                  Color.fromARGB(255, 255, 0, 0),
                                  Color.fromARGB(255, 0, 0, 255),
                                ],
                                    orientation: GradientOrientation.leftDiagonal))*/

                        )),
```

results in the following qr code

![image](https://user-images.githubusercontent.com/17147924/205941441-d00ec571-04df-41b9-a4e8-96370578eadf.png)

looks like since the latest flutter upgrade the qr ball is faulted

I am working on a Flutter Web project with canvas renderer!

**Comment by alexzhirkevich** [comment]
Looks like a dart.ui.Path bug. The same thing was on mobile, and was fixed just by combining resulting path with empty one:

https://github.com/alexzhirkevich/custom-qr-generator-flutter/blob/a8626e5b5c0e15dfc46f9cd0dfdcf0c012b3c621/lib/qr_painter.dart#L63-L66

Unfortunately, that didn't solve the problem on web.

**Comment by alexzhirkevich** [comment]
> Unfortunately, that didn't solve the problem on web.

This fix is already in released version. It works for mobile but not for web. I'll try to fix it a little bit later.

**Comment by razfazz** [comment]
ok thank you. and also thank you for that great package!

**Comment by zogol** [comment]
Much better approach then the other package eg barcode package. I am building a crossplatform app which uses qr codes at specific places and would switch right away when the bug on web is fixed.
I thank you too.

**Comment by alexzhirkevich** [comment]
Fixed in 0.1.2

**Comment by razfazz** [comment]
thank you very much!
